server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -804,10 +804,12 @@ public TSDataType getSeriesType(PartialPath path) throws MetadataException {
 
   public MeasurementMNode[] getMNodes(PartialPath deviceId, String[] measurements)
       throws MetadataException {
-    MNode deviceNode = getNodeByPath(deviceId);
     MeasurementMNode[] mNodes = new MeasurementMNode[measurements.length];
     for (int i = 0; i < mNodes.length; i++) {
-      mNodes[i] = ((MeasurementMNode) deviceNode.getChild(measurements[i]));
+      try {
+        mNodes[i] = (MeasurementMNode) getNodeByPath(deviceId.concatNode(measurements[i]));
+      } catch (PathNotExistException ignored) {
+      }
```

Comment:
hi,  I have a minor..
If an exception occurs, a line of logs must be recorded to facilitate fault tracing.

Reply:
OK